### ISSUE TITLE
Extensions not found when using decorated annotations

### ISSUE THREAD

**Issue by robzienert** [pf4j/pf4j#347]
It looks like PF4J's `@Extension` annotation cannot be put onto another annotation.

My project is looking to add additional metadata and PF4J does not appear to find any extensions that use our `@SpinnakerExtension` annotation:

```java
@Extension
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Documented
public @interface SpinnakerExtension {
}
```

The `SpinnakerExtension` annotation has additional fields which are required for our project to process things correctly. The workaround for the moment, is to include both annotations on an extension, but this is undesirable as a long-term solution.

**Comment by decebals** [comment]
If I understand correctly, this issue is identical with the feature describes [here](https://github.com/pf4j/pf4j/issues/182#issuecomment-377484061).

**Comment by decebals** [comment]
@robzienert, @tylerhawkes
I think that I solved this feature request (see PR #348). I published a new **SNAPSHOT** that contains this feature.
Can you test it please?
If you use Maven in your project(s), please take a look at [doc](https://pf4j.org/dev/maven.html) to find how to use **SNAPSHOT** in your project(s).

**Comment by decebals** [comment]
@robzienert @tylerhawkes 
My intention is to release a new version soon, that mainly will contain this feature and #289. In this context, do you think this problem is solved (with what we have on the master's branch) or do you expect more?

**Comment by jonsie** [comment]
@decebals This release looks like it will achieve the original request here - thank you!

**Comment by decebals** [comment]
This feature is included in the latest release (3.2.0).
